rfcs/proposed/zip_view/README.md

```diff
@@ -0,0 +1,40 @@
+# zip_view Support for the oneDPL Range APIs with C++20
+
+## Introduction
+`std::ranges::zip_view` is a powerful utility that enables developers to combine two or more ranges into a single view,
+where each element is represented as a tuple containing corresponding elements from each input range.
+
+## Motivations
+`std::ranges::zip_view` is a convenient way to combine multiple ranges into a single view, where each element of
+the resulting range is a tuple containing one element from each of the input ranges. This can be particularly
+useful for iterating over multiple collections in parallel. `std::ranges::zip_view` was introduced in C++23,
+but many developers are still using C++20 standard. So, oneDPL introduces `oneapi::dpl::ranges::zip_view`,
```

Comment:
Is this feature going to be experimental? If yes, I guess the namespace should be  `oneapi::dpl::experimental::ranges::zip_view`.

Reply:
Firstly, it is going to be experimental, yes.
Regarding to `experimental` namespace here. I don't know... It depends on RFC process. This document is a proposal in the product or to be productized. So, `experimental` word looks weird.

Talking about the next the next stage of our doing with `zip_view`, yes we would like to have it in the experimental namespace first.

Reply:
I guess you expect the feature to be productized fast, similarly to the C++20 range support. Hence, having `experimental` namespace will impair user experience.

Does it make sense to treat is as an extension straight away? Do you feel confident regarding the current design?

Reply:
My understanding here was that we were shooting for full product status for the ranges MVP APIs including `zip_view` after the experience of the existing experimental ranges APIs.  We already have an experimental `zip_view` that has been around for some time now.

Reply:
Ah, indeed, thanks for reminding about  `oneapi::dpl::experimental::ranges::zip_view`. Then, "supported" stage looks better than "experimental" as the next target.


> Talking about the next stage of our doing with zip_view, yes we would like to have it in the experimental namespace first.

Could you elaborate why you prefer "experimental" stage?

Reply:
> Could you elaborate why you prefer "experimental" stage?

We don't have `zip_view` in the oneDPL spec yet... It will take additional time to prepare it. But we have very useful ready implemented utility right know. And we would like to have one in the closest oneDPL release.
Next, as far as the spec is ready, we will move `zip_view` from experimental to the product.

Reply:
But, talking about existing `oneapi::dpl::experimental::ranges::zip_view`, not compliant with the standard `zip_view`... yes, that proposal rather should explain co-existance of these two implementations, I agree.

Reply:
Basically,  there is an internal utility - `oneapi::dpl::__ranges::zip_view` in using everywhere in the internal oneDPL implementation with C++17 support.
Also `oneapi::dpl::__ranges::zip_view` is injected into `oneapi::dpl::experimental::ranges` namespace. 
`oneapi::dpl::experimental::ranges::zip_view` is used in 4 places in oneDPL impl... and probably in a user'code, because it is a public name.

So, I would propose for `_ONEDPL_CPP20_RANGES_PRESENT` mode to have the proposed new `zip_view` in the `oneapi::dpl::experimental::ranges` namespace and  injection of existing `oneapi::dpl::__ranges::zip_view` into `oneapi::dpl::experimental::ranges` otherwise.

@akukanov , @danhoeflinger , @dmitriy-sobolev what do you think?

Reply:
In theory, we have license to change "experimental" things, as you suggest (in C++20 in this case).

What we need to be careful of is that the existing `zip_view` has been stable for a long time, and I think used in non-experimental production code internally.  I think we need to confirm and be sure that we are not introducing breaking changes into non-experimental code, and that any breaking changes to the experimental zip_view with C++20 enabled are well described and kept to a minimum.  

As far as the RFC is concerned, we need to make it clear here the plan for how it will coexist with what already exists (basically what you describe above).

Reply:
> the existing `zip_view` has been stable for a long time

The existing zip_view doesn't work with C++ standard ranges. That's why we need new one  for _ONEDPL_CPP20_RANGES_PRESENT mode.

Reply:
Yes that makes sense.  My reason for stating its stability is not to indicate that a new version isn't warranted, but merely that users may have more of an expectation to treat the existing implementation as a "supported" feature rather than an "experimental" feature, because we have basically treated it as such.  
Therefore, changes may take them by surprise.  We are not doing anything wrong by changing it, as it is experimental, we just need to be aware that it may affect users.